onnx/inliner.py

```diff
@@ -28,7 +28,7 @@ def inline_local_functions(
 
 
 def inline_selected_functions(
-    model: onnx.ModelProto, function_ids: list[tuple[str, str]], exclude: bool = False
+    model: onnx.ModelProto, function_ids: list[tuple[str, str]], exclude: bool = False, inline_schema_functions: bool = False
```

Comment:
I wonder if we should set it to true because when a user provides the function ID, I imagine they want the function to be inlined (at the trade off of a modified behavior when exclude=False).

Reply:
Not sure, worth thinking over. But there is a plausible scenario where even schema-defined functions can appear as model-local functions, and they are applied first. So, ("", "Softmax") can also be a model-local function.

Reply:
Some follow up:

- Are local functions allowed to alias schema functions in general? 
- What happens when `exclude = True` and `inline_schema_functions = True`? Do all schema functions get inlined? In this case we should keep the default to `False` to preserve previous behavior

Reply:
Is `function_ids` referring to both model-local functions and schema-defined functions here? 
For a schema-defined function, is id referring to the name or op_type? "Softmax" would be the op_type, right?

Reply:
Yes, local functions can alias schema. Yes, function_ids can refer to both. Function-ids refer to the operator name. (Yuan, are you asking about NodeProto.name, which is a node-label? Yes, that is different, and we mean the op_type within a NodeProto.)

Reply:
> * What happens when `exclude = True` and `inline_schema_functions = True`? Do all schema functions get inlined?

Yes, that's correct. Well, what `exclude=True` means is "inline all functions NOT explicitly listed here", and that includes schema-defined functions if `inline_schema_functions = True`

Reply:
Thanks for clarifying.
Another question: are nested function schemas expanded by default like local functions? ie, when the function schema of one op makes use of other function ops. Maybe add some more test cases to illustrate this point? I don't think we'll have any cycles in the function schemas, but maybe worth guarding against it with a max recursive depth?

Reply:
Yes, nested calls will be expanded (as long as they match the filter user provides). There is already a test-case to illustrate this: SoftmaxCrossEntropyLoss is defined in terms of LogSoftmax, which is also a function. Please see `TEST(SchemaFunctionInliner, NestedTest)`. ONNX functions are NOT intended to be recursive.